packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m

```diff
@@ -174,15 +174,15 @@ - (NSDictionary *)encodingMap
 }
 
 EX_EXPORT_METHOD_AS(getInfoAsync,
-                    getInfoAsyncWithURI:(NSString *)uriString
+                    getInfoAsyncWithURI:(NSString *)pathString
                     withOptions:(NSDictionary *)options
                     resolver:(EXPromiseResolveBlock)resolve
                     rejecter:(EXPromiseRejectBlock)reject)
 {
-  NSURL *uri = [NSURL URLWithString:uriString];
+  NSURL *uri = [NSURL URLWithString:[pathString stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]]];
```

Comment:
Since we are using `[pathString stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]]` all over the place, maybe it would be good to refactor this out to an internal method

Reply:
As I proposed at the end of this comment https://github.com/expo/expo/pull/21098#pullrequestreview-1285860715, probably the best way would be to change this convertible type and migrate other packages using this to the new API 😄 The internal code of packages can stay in Objective-C if migrating it to Swift would be too much time consuming. I'm encouraging everyone to help in migrating them because only then we could drop a lot of legacy code from expo-modules-core.